docs/DeveloperGuide.md

```diff
@@ -1,18 +1,616 @@
+| ```Logic``` |Outlines the abstract behavior of commands, as well as handle verification of params with appropriate error handling.
+
+
+
+##### Command and Logic
+
+![](uml_images/manualTracker/images/Commands_Logic_edited.png)
+
+|Class| Function |
+|--------|----------|
+|```retrieveLedgerCommand```| Process ```paramTypes```-```param``` pairs from the ```CommandPacket``` instance to identify specified ```Ledger``` instance, then retrieves the instance from the existing ```LedgerList```.
+|```createLedgerCommand```| Process ```paramTypes```-```param``` pairs from the ```CommandPacket``` instance to identify specified ```Ledger``` instance to be created, then creates the instance and append to existing ```LedgerList```.
+|```retrieveEntryCommand```| Omitted and left as exercise for reader. : ^ )
+|```createEntryCommand```| Omitted for brevity.
+|```editEntryCommand```| Omitted for brevity.
+|```ParamChecker```| Class contains a collection of methods that verify the correctness of the ```param``` supplied. <br><br> For instance, ```ParamChecker.checkAndReturnIndex``` checks if the index provided is out of bounds relative to the specified list, and throws the relevant exception if the input index is invalid. 
+|```ParamHandler```| Abstract class that outlines the general param handling behavior of ```commands``` instances and other classes that need to handle ```params``` in its operation.  
+
+##### Handler and Command
+
+![](uml_images/manualTracker/images/Handler_Commands.png)
+
+|Class| Function |
+|--------|----------|
+|```retrieveLedgerCommand```| Refer to section above.
+|```createLedgerCommand```| Refer to section above.
+|```retrieveEntryCommand```| Omitted for brevity.
+|```createEntryCommand```| Omitted for brevity.
+|```editEntryCommand```| Omitted for brevity.
+|```ManualTracker```| Implements Manual Tracker. Contains handler methods that implements a particular operation capable by the Manual Tracker. <br><br> These methods use the above ```command``` instances for param handling operations from user input.
+|```EntryTracker```| Omitted for brevity.
+
+##### Handler and Parser
+
+![](uml_images/manualTracker/images/Handler_Parser.png)
+
+|Class| Function |
+|--------|----------|
+|```InputParser```| Breaks input string by user into ```commandString``` and a sequence of ```paramTypes```-```param``` pairs. <br><br> The latter subsequence of the string is passed into ParamParser for further processing. <br><br> Information obtained from input parsing will be used to populate an instantiated ```CommandPacket``` instance, which will then be passed to the entity that called the parsing function.
+|```ParamParser```| Process the sequence of ```paramTypes```-```param``` pairs and populate the ```paramMap``` in the instantiated ```CommandPacket``` instance.
+|```ManualTracker```| Refer to section above.
+|```EntryTracker```| Omitted for brevity.
+
+##### Handler and Data
+
+![](uml_images/manualTracker/images/Handler_Data.png)
```

Comment:
I think a better option might be to put the aggregation symbol from LedgerList to Ledger and EntryList to Entry (by putting a shaded diamond at LedgerList in the arrow pointing from LedgerList to Ledger).